### DISCUSSION TITLE
auto-checkout does not fire on file save

### DISCUSSION THREAD

**Discussion by shaharphilips** [nik-base/auto-tfs#14] [General]
Hi,

I am using Auto TFS v1.3.7 on VS code 1.94.0. The extension is amazing, but I have one critical issue.
On one of my PCs the auto-checkout on save doesnt work. I have to use the on-save option because the on-change option causes a significant performance issue. When I try to save the file, VS Code complains that it is read only, and it seems like Auto TFS was not triggered and there was no attempt to checkout the file.
What should I try to work around the issue? It seems to be local to one of my PCs.

I have tried a complete uninstall and reinstall of VS code but it did not help.
Checkout "on channges" seems to work well, but has significant performance impact.

This is my AutoTFS config
   "auto-tfs.tfs.autoCheckout": "On Save",
    "auto-tfs.tfs.autoDelete": true,
    "auto-tfs.tfs.autoRename": true,
    "auto-tfs.tf.path": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Professional\\Common7\\IDE\\CommonExtensions\\Microsoft\\TeamFoundation\\Team Explorer\\TF.exe",

**Comment by nik-base** [comment]
Hi @shaharphilips ,

Could you please try a couple of things -

- When you save the file, if it does not appear in source control. Try clicking on `sync` button and check if it appears now.
- You can also try to enable `Tfs: Confirm` setting of AutoTFS extension and check if file is getting checked out.

Let me know if any of this works. If not, please enable `Tfs: Debug` and share the logs from AutoTFS output channel.

**Comment by shaharphilips** [comment]
Thank you for your help.

- The checked-out files appear on the source control even before I sync them.
- **Important** - When VS code complains the file is read-only, if I press "Override", once the overide is done, Auto TFS DOES checkout the file properly. It seems like only the override triggerd Auto TFS and not the initial save.
- setting TFS: Confirm did not change the behaviour.
- outoput of Tfs:Debug

Getting info on c:\tfs\Dev\source\Web\client\client\src\common\ui\components\favorite-button\cp-favorite-button.template.html from source control
TF command 'info' is in progress...
Local information:
  Local path : C:\tfs\Dev\source\Web\client\client\src\common\ui\components\favorite-button\cp-favorite-button.template.html
  Server path: $/RISTeam/SourceCode/Dev/source/Web/client/client/src/common/ui/components/favorite-button/cp-favorite-button.template.html
  Changeset  : 39481
  Change     : none
  Type       : file
Server information:
  Server path  : $/RISTeam/SourceCode/Dev/source/Web/client/client/src/common/ui/components/favorite-button/cp-favorite-button.template.html
  Changeset    : 39481
  Deletion ID  : 0
  Lock         : none
  Lock owner   :
  Last modified: Thursday, February 9, 2023 3:56:27 PM
  Type         : file
  File type    : Windows-1252
  Size         : 940
 
TF command 'info' is successfully done
none
TF command 'checkout' is in progress...

**Comment by nik-base** [comment]
Hi @shaharphilips ,

The checkout is triggered on Save. Since your file is readonly, no save file event is triggered until you click on 'Overwrite'. So you have to always click on overwrite for checkout to work.

It seems like you are using Server workspace mode in TFS, which marks files which are not checked out readonly. You can read more here [All files are marked as read only after check in to TFS from Visual Studio 2012](https://stackoverflow.com/questions/19855851/all-files-are-marked-as-read-only-after-check-in-to-tfs-from-visual-studio-2012).

Consider changing your workspace to local workspace.

Hope it helps!

**Comment by shaharphilips** [comment]
OK. Got it.
Thanks for all your help!
